controller/engine_controller.go

```diff
@@ -1790,28 +1787,6 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replicaName, add
 			}
 		}
 
-		// It has been at least 10 seconds (2 * EnginePollInterval) since we were asked to rebuild.
-		// Should we still communicate with addr?
-		// TODO: Remove this check in versions that include a completed longhorn/longhorn#5845.
-		updatedEngine, err := ec.ds.GetEngineRO(e.Name)
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				log.Error("Failed to proceed to rebuild since engine is not found")
-				return
-			}
-			log.WithError(err).Error("Failed to get latest engine spec before rebuilding")
-			return
-		}
-		updatedAddr, ok := updatedEngine.Spec.ReplicaAddressMap[replicaName]
-		if !ok {
-			log.Warnf("Refusing to rebuild since replica %v is no longer in the replicaAddressMap", replicaName)
-			return
-		}
-		if addr != updatedAddr {
-			log.Warnf("Refusing to rebuild since the address for replica %v has been updated from %v to %v", replicaName, addr, updatedAddr)
-			return
-		}
-
```

Comment:
The commit moves the them to the earlier place and only checks the conditions when `autoCleanupSystemGeneratedSnapshot` is enabled. Should we check them regardless of the value of `autoCleanupSystemGeneratedSnapshot`?

Reply:
I am fine with it since **rebuilding** always requires that.

Reply:
My thinking is that the it is the 10 second snapshot purge wait that causes us the largest danger. If `autoCleanupSystemGenerateSnapshot = false`, it should be relatively little time between the start of an engine reconciliation and the moment we kick of a rebuild. Even if we do check, we can never be 100% sure the situation hasn't changed in the milliseconds between to lines of execution, so I confined the check to the situation we know can cause us problems.

I don't feel particularly strongly that we shouldn't also check when `autoCleanupGenerateSnapshot = false` if that's what the team wants. I just think the code is cleaner this way.